### DISCUSSION TITLE
Integer division

### DISCUSSION THREAD

**Discussion by pd-giz-dave** [vlang/v#19126] [Ideas]
I'm just getting started using V and I like it, but there is an issue that keeps tripping me up and that is division of integers yields an integer, reasonable but dangerous in the following context:

width := 100
height := 50
ratio := height / width  // result is an int of 0
expect := 0.5  // an f32
if ratio == expect { println('ok')} else { println('fail')}  // fails
println(ratio)

I would expect f32 == int to fail at compile time as a type mismatch

How about a new operator '\' meaning coerce operands to float and generate a float answer?

**Comment by JalonSolov** [comment]
The compiler did exactly as you asked.  If you want it to do something different, ask it to do that.
```v
width := 100
height := 50
ratio := f32(height) / width // result is a float of 0.5
expect := 0.5 // an f32
if ratio == expect { println('ok')} else { println('fail')} // succeeds
println(ratio)
```

**Comment by pd-giz-dave** [comment]
My point is that I made an error but it was 'hidden' by the float to integer corecion in the ==. IMO auto coerecion between integer and float is dangerous because it can lead to unexpected behaviour as in my very simple example. If the compiler failed in my eg then it would trigger a correction to your example. Without such a hint the error would go undetected until run-time if at all.

**Reply by JalonSolov** [reply]
This is very common in programming languages.  Very few languages will complain rather than simply doing what you ask.

"[#](https://calcitfast.com/integer-division#what-is-an-integer-division) What is an integer division?

Integer Division is defined as the division in which the remainder (fractional part) is simply dropped or discarded, no matter how big it is. Thus, the resultant value of an Integer division is always an integer."

**Reply by pd-giz-dave** [reply]
One of the goals of V is to be 'safe', I posit that auto coercion of integer to float is not safe as my example illustrates.

**Reply by JalonSolov** [reply]
It is perfectly safe, as it is working as designed.  It may be unexpected if you are unfamiliar with the way integer division works, but it is not unsafe.
